rumqttd/src/router/routing.rs

```diff
@@ -365,12 +384,35 @@ impl Router {
         let span = tracing::info_span!("incoming_disconnect", client_id);
         let _guard = span.enter();
 
+        // must handle last will before sending disconnect packet
+        // as the disconnecting client might have subscribed to will topic.
         if execute_last_will {
             self.handle_last_will(id);
         }
 
         info!("Disconnecting connection");
 
+        if let Some(reason_code) = reason {
```

Comment:
if i understand this correctly, the invariant here is if reason is `Some(reason_code)` then we assume its a v5 client and we send a disconnect packet to it right?

Reply:
Yes. We can say so for now, more in-depth work in required for disconnection related feature. But beyond the scope of this PR.